### ISSUE TITLE
imbalanced classification

### ISSUE THREAD

**Issue by shahlaebrahimi** [elkasoapy/google_codejam#1]
Dear Dr. Lopez

Greetings

I studied your informative article with the title "an insight into classification with imbalanced data: empirical results and current trends on using data intrinsic characteristics". 

I would greatly appreciate if you could let me know if the class-weight in the [sklearn's classifiers e.g., logistic regression ](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/linear_model/logistic.py) (especially when it is set as dict) is classified into direct method cost-sensitive or meta-learning cost-sensitive?

I mean based on the following categorization which you provided in your article, how it should be categorized?

> In summary, two main general approaches have been proposed to deal with cost-sensitive problems:
> 
> **1. Direct methods:** The main idea of building a direct cost-sensitive learning algorithm is to directly introduce and utilize misclassification costs into the learning algorithms. For example, in the context of decision tree induction, the tree-building strategies are adapted to minimize the misclassification
>  costs. The cost information is used to: (1) choose the best attribute to split the data and (2)
>  determine whether a subtree should be pruned. On the other hand, other approaches based on genetic algorithms can incorporate misclassification costs in the fitness function.
> 
> **2. Meta-learning:** This methodology implies the integration of a ‘‘preprocessing’’ mechanism for the training data or a ‘‘postprocessing’’ of the output, in such a way that the original learning algorithm is not modified. Cost-sensitive meta-learning can be further classified into two main categories: thresholding and sampling:
> **- Thresholding** is based on the basic decision theory that assigns instances to the class with minimum expected cost. For example, a typical decision tree for a binary classification problem assigns the class label of a leaf node depending on the majority class of the training samples that reach the node. A cost-sensitive algorithm assigns the class label to the node that minimizes the classification cost.
> 
> **- Sampling** is based on modifying the training dataset. The most popular technique lies in resampling the original class distribution of the training dataset according to the cost decision matrix by means of undersampling/oversampling or assigning instance weights. These modifications have shown to be effective and can also be applied to any cost insensitive learning algorithm.
> 

Thanks in advance.

Best regards,
